### PR TITLE
api (ticdc): check http request context in api helper

### LINKED ISSUE #11046: The Error message return by create changefeed API  was misleading.
### What did you do?

Create a changefeed sink to GCS. 

### What did you expect to see?

Changefeed is created successfully. 

### What did you see instead?

An error return.
```
[2024/04/30 14:23:56.490 +00:00] [INFO] [middleware.go:49] [/api/v2/changefeeds] [status=500] [method=POST] [path=/api/v2/changefeeds] [query=] [ip=[172.16.19.194](http://172.16.19.194/)] [user-agent=Go-http-client/1.1] [client-version=] [error="[DFLOW:ErrFailToCreateExternalStorage]failed to create external storage%!(EXTRA string=creating ExternalStorage for s3): context canceled"] [errorVerbose="[DFLOW:ErrFailToCreateExternalStorage]failed to create external storage%!(EXTRA string=creating ExternalStorage for s3): context canceled\[ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20221009092201-b66cddb77c32/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20221009092201-b66cddb77c32/normalize.go:164\ngithub.com/pingcap/tiflow/pkg/util.GetExternalStorage\n\tgithub.com/pingcap/tiflow/pkg/util/external_storage.go:84\ngithub.com/pingcap/tiflow/pkg/util.GetExternalStorageFromURI\n\tgithub.com/pingcap/tiflow/pkg/util/external_storage.go:42\ngithub.com/pingcap/tiflow/cdc/sink/dmlsink/cloudstorage.NewDMLSink\n\tgithub.com/pingcap/tiflow/cdc/sink/dmlsink/cloudstorage/cloud_storage_dml_sink.go:110\ngithub.com/pingcap/tiflow/cdc/sink/dmlsink/factory.New\n\tgithub.com/pingcap/tiflow/cdc/sink/dmlsink/factory/factory.go:105\ngithub.com/pingcap/tiflow/cdc/sink/validator.Validate\n\tgithub.com/pingcap/tiflow/cdc/sink/validator/validator.go:55\ngithub.com/pingcap/tiflow/cdc/api/v2.APIV2HelpersImpl.verifyCreateChangefeedConfig\n\tgithub.com/pingcap/tiflow/cdc/api/v2/api_helpers.go:236\ngithub.com/pingcap/tiflow/cdc/api/v2.(*OpenAPIV2).createChangefeed\n\tgithub.com/pingcap/tiflow/cdc/api/v2/changefeed.go:109\ngithub.com/gin-gonic/gin.(*Context).Next\n\tgithub.com/gin-gonic/gin@v1.8.1/context.go:173\ngithub.com/pingcap/tiflow/cdc/api/v2.RegisterOpenAPIV2Routes.ForwardToControllerMiddleware.func4\n\tgithub.com/pingcap/tiflow/cdc/api/middleware/middleware.go:96\ngithub.com/gin-gonic/gin.(*Context).Next\n\tgithub.com/gin-gonic/gin@v1.8.1/context.go:173\ngithub.com/pingcap/tiflow/cdc/api/v2.RegisterOpenAPIV2Routes.ErrorHandleMiddleware.func3\n\tgithub.com/pingcap/tiflow/cdc/api/middleware/middleware.go:65\ngithub.com/gin-gonic/gin.(*Context).Next\n\tgithub.com/gin-gonic/gin@v1.8.1/context.go:173\ngithub.com/pingcap/tiflow/cdc/api/v2.RegisterOpenAPIV2Routes.LogMiddleware.func2\n\tgithub.com/pingcap/tiflow/cdc/api/middleware/middleware.go:39\ngithub.com/gin-gonic/gin.(*Context).Next\n\tgithub.com/gin-gonic/gin@v1.8.1/context.go:173\ngithub.com/pingcap/tiflow/cdc/api/v2.RegisterOpenAPIV2Routes.CheckServerReadyMiddleware.func1\n\tgithub.com/pingcap/tiflow/cdc/api/middleware/middleware.go:179\ngithub.com/gin-gonic/gin.(*Context).Next\n\tgithub.com/gin-gonic/gin@v1.8.1/context.go:173\ngithub.com/gin-gonic/gin.CustomRecoveryWithWriter.func1\n\tgithub.com/gin-gonic/gin@v1.8.1/recovery.go:101\ngithub.com/gin-gonic/gin.(*Context).Next\n\tgithub.com/gin-gonic/gin@v1.8.1/context.go:173\ngithub.com/gin-gonic/gin.(*Engine).handleHTTPRequest\n\tgithub.com/gin-gonic/gin@v1.8.1/gin.go:616\ngithub.com/gin-gonic/gin.(*Engine).ServeHTTP\n\tgithub.com/gin-gonic/gin@v1.8.1/gin.go:572\nnet/http.serverHandler.ServeHTTP\n\tnet/http/server.go:2938\nnet/http.(*conn).serve\n\tnet/http/server.go:2009\nruntime.goexit\n\truntime/asm_amd64.s:1650](http://ngithub.com/pingcap/errors.AddStack/n/tgithub.com/pingcap/errors@v0.11.5-0.20221009092201-b66cddb77c32/errors.go:174/ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs/n/tgithub.com/pingcap/errors@v0.11.5-0.20221009092201-b66cddb77c32/normalize.go:164/ngithub.com/pingcap/tiflow/pkg/util.GetExternalStorage/n/tgithub.com/pingcap/tiflow/pkg/util/external_storage.go:84/ngithub.com/pingcap/tiflow/pkg/util.GetExternalStorageFromURI/n/tgithub.com/pingcap/tiflow/pkg/util/external_storage.go:42/ngithub.com/pingcap/tiflow/cdc/sink/dmlsink/cloudstorage.NewDMLSink/n/tgithub.com/pingcap/tiflow/cdc/sink/dmlsink/cloudstorage/cloud_storage_dml_sink.go:110/ngithub.com/pingcap/tiflow/cdc/sink/dmlsink/factory.New/n/tgithub.com/pingcap/tiflow/cdc/sink/dmlsink/factory/factory.go:105/ngithub.com/pingcap/tiflow/cdc/sink/validator.Validate/n/tgithub.com/pingcap/tiflow/cdc/sink/validator/validator.go:55/ngithub.com/pingcap/tiflow/cdc/api/v2.APIV2HelpersImpl.verifyCreateChangefeedConfig/n/tgithub.com/pingcap/tiflow/cdc/api/v2/api_helpers.go:236/ngithub.com/pingcap/tiflow/cdc/api/v2.(*OpenAPIV2).createChangefeed/n/tgithub.com/pingcap/tiflow/cdc/api/v2/changefeed.go:109/ngithub.com/gin-gonic/gin.(*Context).Next/n/tgithub.com/gin-gonic/gin@v1.8.1/context.go:173/ngithub.com/pingcap/tiflow/cdc/api/v2.RegisterOpenAPIV2Routes.ForwardToControllerMiddleware.func4/n/tgithub.com/pingcap/tiflow/cdc/api/middleware/middleware.go:96/ngithub.com/gin-gonic/gin.(*Context).Next/n/tgithub.com/gin-gonic/gin@v1.8.1/context.go:173/ngithub.com/pingcap/tiflow/cdc/api/v2.RegisterOpenAPIV2Routes.ErrorHandleMiddleware.func3/n/tgithub.com/pingcap/tiflow/cdc/api/middleware/middleware.go:65/ngithub.com/gin-gonic/gin.(*Context).Next/n/tgithub.com/gin-gonic/gin@v1.8.1/context.go:173/ngithub.com/pingcap/tiflow/cdc/api/v2.RegisterOpenAPIV2Routes.LogMiddleware.func2/n/tgithub.com/pingcap/tiflow/cdc/api/middleware/middleware.go:39/ngithub.com/gin-gonic/gin.(*Context).Next/n/tgithub.com/gin-gonic/gin@v1.8.1/context.go:173/ngithub.com/pingcap/tiflow/cdc/api/v2.RegisterOpenAPIV2Routes.CheckServerReadyMiddleware.func1/n/tgithub.com/pingcap/tiflow/cdc/api/middleware/middleware.go:179/ngithub.com/gin-gonic/gin.(*Context).Next/n/tgithub.com/gin-gonic/gin@v1.8.1/context.go:173/ngithub.com/gin-gonic/gin.CustomRecoveryWithWriter.func1/n/tgithub.com/gin-gonic/gin@v1.8.1/recovery.go:101/ngithub.com/gin-gonic/gin.(*Context).Next/n/tgithub.com/gin-gonic/gin@v1.8.1/context.go:173/ngithub.com/gin-gonic/gin.(*Engine).handleHTTPRequest/n/tgithub.com/gin-gonic/gin@v1.8.1/gin.go:616/ngithub.com/gin-gonic/gin.(*Engine).ServeHTTP/n/tgithub.com/gin-gonic/gin@v1.8.1/gin.go:572/nnet/http.serverHandler.ServeHTTP/n/tnet/http/server.go:2938/nnet/http.(*conn).serve/n/tnet/http/server.go:2009/nruntime.goexit/n/truntime/asm_amd64.s:1650)"] [duration=10.08116892s]
```

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
v7.1.4
```

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: close #11046 

#### Tests 

 - Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/api/v2/api_helpers.go [hunk 1]
```diff
@@ -551,5 +579,9 @@ func (h APIV2HelpersImpl) getVerifiedTables(
 		return nil, nil, err
 	}
 
+	if ctx.Err() != nil {
```
Comment (CharlesCheung96) [inline cdc/api/v2/api_helpers.go:582]: Perhaps the redundant code can be reduced through a common `ctxChecker` function? eg:
```go
func ctxChecker(ctx, fn func() error) {
    if fn() != nil {
		return nil, errors.Trace(err)
	}
	if ctx.Err() != nil {
		return nil, errors.Trace(ctx.Err())
	}
}
```

**Review by sdojjy — APPROVED** [review]

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**